tendermint-testgen/src/vote.rs

```diff
@@ -0,0 +1,165 @@
+use gumdrop::Options;
+use serde::Deserialize;
+use signatory::{
+    ed25519,
+    signature::{Signature as _, Signer},
+};
+use simple_error::*;
+use tendermint::{block, lite, signature::Signature, vote, Time};
+
+use crate::{helpers::*, Generator, Header, Validator};
+
+#[derive(Debug, Options, Deserialize, Clone)]
+pub struct Vote {
+    #[options(
+        help = "validator of this vote (required; can be passed via STDIN)",
+        parse(try_from_str = "parse_as::<Validator>")
+    )]
+    pub validator: Option<Validator>,
+    #[options(help = "validator index (default: from commit header)")]
+    pub index: Option<u64>,
+    #[options(help = "header to sign (default: commit header)")]
+    pub header: Option<Header>,
+    #[options(help = "vote type; 'precommit' if set, otherwise 'prevote' (default)")]
+    pub precommit: Option<()>,
```

Comment:
Since `gumdrop` supports boolean options, could you explain the choice of `Option<()>` instead?

Reply:
The alternatives are:
* using plain `bool`: doesn't fit well with other fields being options, so contradicts the overall architecture.
* using `Option<bool>` doesn't work either, because then `gumdrop` expects the input of the form `--precommit true` or `--precommit false`.

Using `Option<()>` on the other hand, allows in the CLI just to write `--precommit`.